### PR TITLE
Fix small issues pointed out by static analysis

### PR DESCRIPTION
**Pull request by visuve**
Was just checking out the project out of curiosity and ran some static analysis. Maybe I will commit something more useful later...

### CODE REVIEW

File: kernel/src/kalloc.cpp [hunk 1]
```diff
@@ -580,7 +580,7 @@ void kalloc::debug(){
 
     k_print_line();
 
-    printf("memory free in malloc chain: %m (%u)\n", memory_free, memory_free);
-    printf("There are %u non free blocks in the free list\n", non_free_blocks);
-    printf("There are %u inconsistent sized blocks in the free list\n", inconsistent);
+    printf("memory free in malloc chain: %m (%zu)\n", memory_free, memory_free);
+    printf("There are %zu non free blocks in the free list\n", non_free_blocks);
+    printf("There are %zu inconsistent sized blocks in the free list\n", inconsistent);
```
Comment (wichtounet) [inline kernel/src/kalloc.cpp]: While correct for standard printf, this is not correct here. I am using my own implementation of printf and I have not implemented zu yet.

**Review by wichtounet — CHANGES_REQUESTED** [review]
Hi, 

Thanks for your changes 👍 

Can you just revert the changes to printf? Check my comment for why.

I will merge the other changes once you did it.

File: kernel/src/kalloc.cpp (hunk 1 above)
Comment (visuve) [reply kernel/src/kalloc.cpp]: Done, sorry it took so long.

**Review by wichtounet — APPROVED** [review]

**Comment by wichtounet** [comment]
Thanks for the changes. 
Don't worry about the time! It took me quite a while to review your changes too ;)

**Pull request merged**